### ISSUE TITLE
It is not clear why templates with a default mapping created prior to 7.x fail in 7.x

### ISSUE THREAD

**Issue by talevy** [elastic/elasticsearch#48427]
**Problem**

It is not clear how to resolve templates with default mappings in 7.x.

Indices created in 7.x matching templates created with the default mapping in 6.x fail to create
with following exception:

```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "The [default] mapping cannot be updated on index [my_index]: defaults mappings are not useful anymore now that indices can have at most one type."
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "The [default] mapping cannot be updated on index [my_index]: defaults mappings are not useful anymore now that indices can have at most one type."
  },
  "status" : 400
}
```

This error makes it clear that there is something wrong with the mapping, which results in users exploring templates for default-mapping usage. The problem is that users will not see their templates defining the `_default_` mapping since 7.x defaults to the typeless APIs. So `include_type_name=true` must be set on the GET Templates API to see this information. 

**A Few Possible Solutions**

- specify how to take action in the error message
- document this breaking change better in the docs
- add better template upgrade support in the Upgrade Assistant (outside scope of this repository)

**Comment by praveenmak** [comment]
I am seeing the same error after upgrading to 7.4.2 from 6.4.2

I get this when I do a rollover. I am using alias and we rollover to new one using URL math.

May I know what is the fix ?

**Comment by freakingid** [comment]
Greetings. I ran in to this as well. Please see:
[Goodbye, types. Hello, typeless](https://www.elastic.co/blog/moving-from-types-to-typeless-apis-in-elasticsearch-7-0)

**Under "Stop using _default_ mappings"**

> You should update your index templates and your application code to add these mappings to the actual type, rather than include them in a _default_ mapping. If the name of the type is not important for you, the recommendation is to use _doc, which will make the transition to typeless APIs easier in the future.

**Comment by jeethsam** [comment]
I ran into the same issue. Is there a fix for this apart from 'not using default mapping'?

**Comment by jtibshirani** [comment]
As @freakingid mentioned, in 7.x you must remove the default mapping from index templates. No other fix is necessary.

To clarify one aspect that can be confusing -- issuing a normal 'get templates' call in 7.x won't show the default mapping, even if it's defined. To see all mappings that are defined, including the default mapping, you should pass the `include_type_name` parameter:

```
GET /_template/my_template?include_type_name
```

**Comment by jtibshirani** [comment]
I merged #54915 to add debugging information to the migration docs. It also updates the error message to link to those docs.
